### PR TITLE
[UNTESTED] [MERGEABLE] Storing in cache and serving

### PR DESCRIPTION
**Pull request by frizinak**

### CODE REVIEW

**Comment by frizinak** [comment]
#8

File: src/EventSubscriber/CacheSubscriber.php [hunk 1]
```diff
@@ -61,22 +87,40 @@ public function onResponse(FilterResponseEvent $event)
...
             }
 
             return;
         }
     }
 
+    public function onEntityPresented(EntityPresentedEvent $event)
+    {
+        $entity = $event->getEntity();
+        $this->presentedEntityTags[] = sprintf(
```
Comment (RobinHoutevelts) [inline src/EventSubscriber/CacheSubscriber.php]:
```php
$this-presentedEntityTags = array_merge($this-presentedEntityTags, $event->getCacheTagsToInvalidate());
```

**Comment by RobinHoutevelts** [comment]
Awesome

File: src/EventSubscriber/CacheSubscriber.php (hunk 1 above)
Comment (frizinak) [inline src/EventSubscriber/CacheSubscriber.php]: The updated

**Comment by frizinak** [comment]
Think it's pretty much done

cache is disabled by default, so we don't break existing sites (i'd bump the minor version number, since we're still in 0.x.y)

will remove the [dont merge] from the title when i've double checked everything, might add it to a single live site before potentially pushing it to all of em.

**Comment by frizinak** [comment]
cache response time was 100ms+, using the ugly HttpKernel as middleware and hooking in way earlier, shaves of about 80ms. The session middleware hasn't kicked in yet at this point, so the 'uid > 0 in session'-check is replaced by a 'is there a session cookie'.

bottleneck atm (with xhprof enabled, so probably slightly skewed) are the autoloader's include calls. Which is pretty good imo.

we should also have a look at the PageCache::lookup method to implement some http request header edge-cases.

**Comment by frizinak** [comment]
well nvm the 'PageCache::lookup' it does what the `RequestPolicy\NoSession` already did 2 calls earlier, ungzips and implements the equivalent of `Symfony\Response::isNotModified()`

wmcontroller:
- [x] session check
- [ ] ungzip `not relevant to wieni, but if this module gets some traction we need to implement it, requires a ini_set so ugly stuff (libraries should have no side effects)`
- [ ] 304 if not modified (as simple as calling isNotModified on the response object)

**Comment by frizinak** [comment]
Should be safe to merge, by default only adds max and smax.

(still need to update readme regarding default page_cache middleware)

**Comment by frizinak** [comment]
Are we all ok with merging?

The only outstanding bug are the active trail issues on 'paged' pages, which I don't believe we'll fix any time soon.

**Pull request merged**